### ISSUE TITLE
Library fails to compile when targeting Linux using a SQLite library that doesn't support snapshots

### ISSUE THREAD

**Issue by dylansturg** [groue/GRDB.swift#1537]
### What did you do?

- Tried to build GRDB on Linux
- Linked with a SQLite distribution built **without** setting `SQLITE_ENABLE_SNAPSHOT` so that various database snapshot related APIs are not available.
- Observed a linker error with unresolved symbols for functions like `sqlite3_snapshot_get`.

I understand that Linux isn't really supported. I _think_ that I'm asking for something relatively small here, and might be able to send a PR.

### What did you expect to happen?

To compile successfully without any undefined symbol references in GRDB code.

### What happened instead?

GRDB referenced undefined symbols that weren't available from the SQLite library, because it was built without enabling `SQLITE_ENABLE_SNAPSHOT`.

### Environment

**GRDB flavor(s):** GRDB
**GRDB version:** v6.24.2
**Installation method:** Manual
**Xcode version:** N/A
**Swift version:** 5.10
**Platform(s) running GRDB:** Linux
**macOS version running Xcode:** N/A

### Demo Project

N/A

### More Context

The relevant code is attempting to guard against this state, as an example, this condition is used in many places throughout the codebase: `#if SQLITE_ENABLE_SNAPSHOT || (!GRDBCUSTOMSQLITE && !GRDBCIPHER && (compiler(>=5.7.1) || !(os(macOS) || targetEnvironment(macCatalyst))))`

Some examples:
- https://github.com/groue/GRDB.swift/blob/4b934fda754b1cab34394ea89c2dd5acd170d50e/GRDB/Core/WALSnapshotTransaction.swift#L2
- https://github.com/groue/GRDB.swift/blob/4b934fda754b1cab34394ea89c2dd5acd170d50e/GRDB/Core/WALSnapshot.swift#L2
- https://github.com/groue/GRDB.swift/blob/4b934fda754b1cab34394ea89c2dd5acd170d50e/GRDB/Core/DatabasePool.swift#L648

When using SQLite built in the same project (but NOT `GRDBCUSTOMSQLITE`) or a pre-built distribution of SQLite on any platform other than macOS, while using compiler newer than 5.7.1, this condition will be true no matter whether that SQLite supports snapshot features. It would be nice to allow disabling these references when using SQLite without database snapshot support on platforms other than macOS.

I'm not positive what's the right behavior here, since most distributions of SQLite probably have database snapshots enabled. Maybe split it so that one of one of `SQLITE_ENABLE_SNAPSHOT || GRDBCUSTOMSQLITE || GRDBCIPHER` is required on Linux?

```
#if SQLITE_ENABLE_SNAPSHOT || (os(Linux) && (GRDBCUSTOMSQLITE || GRDBCIPHER)) || (!GRDBCUSTOMSQLITE && !GRDBCIPHER && (compiler(>=5.7.1) || !(os(macOS) || targetEnvironment(macCatalyst))))
```

That would mean when using a prebuilt SQLite distribution with snapshot support, `SQLITE_ENABLE_SNAPSHOT` would have to be set explicitly to enable the features in GRDB. 

Alternatively, there could be a flag to opt out of the features when the developer knows their SQLite doesn't support snapshots which would be set explicitly as needed when building GRDB in environments like mine. I'm open to that too.

Unfortunately, just setting `SQLITE_ENABLE_SNAPSHOT` when building SQLite isn't an option for me, so it'd be nice if I could build GRDB while excluding the snapshots APIs.

**Comment by groue** [comment]
> This would be acceptable to me. To use snapshots (highly recommended) on Linux, one would have to fork GRDB, and add `SQLITE_ENABLE_SNAPSHOT` to [`swiftSettings`](https://github.com/groue/GRDB.swift/blob/v6.27.0/Package.swift#L7).

Oh, @dylansturg, I'd also warmly welcome your advice on this. I have no experience about the best practices for configuring an SPM package on Linux, so I'll listen to your suggestions (such as env variables, or whatever technique that the community expects to be able to use).

**Comment by vpzomtrrfrt** [comment]
So this would have to be changed in every file it appears in?

**Comment by groue** [comment]
Yes, because the Swift language does not make it possible to define such a condition in a reusable way. Code below is C, not Swift:

```c
#define MY_CONDITION ...
#if MY_CONDITION
...
#endif
```

Now the condition described in this issue was particularly complex in GRDB 6. In GRDB 7, it should be simpler.

```swift
// Original condition in GRDB 6
 #if SQLITE_ENABLE_SNAPSHOT || (!GRDBCUSTOMSQLITE && !GRDBCIPHER && (compiler(>=5.7.1) || !(os(macOS) || targetEnvironment(macCatalyst)))) 

// Original condition in GRDB 7
#if SQLITE_ENABLE_SNAPSHOT || (!GRDBCUSTOMSQLITE && !GRDBCIPHER)
```

**Comment by thinkpractice** [comment]
Small update. Was able to fix linking by adding:

```
#if SQLITE_ENABLE_SNAPSHOT || (!GRDBCUSTOMSQLITE && !GRDBCIPHER && !os(Linux))
```

to all Swift files that had this condition. This also means `SQLITE_ENABLE_SNAPSHOT` should be turned off. Will test if the code really works correctly. Can send a PR, but I guess it would be nice to have a way to enable snapshots on Linux too? Don't understand why the standard `apt` version of `libsqlite3-dev` doesn't have it enabled though.
